pepdbagent/modules/annotation.py

```diff
@@ -0,0 +1,327 @@
+from typing import Union, List
+import logging
+
+from pepdbagent.base_connection import BaseConnection
+from pepdbagent.const import (
+    DEFAULT_LIMIT,
+    DEFAULT_OFFSET,
+    DEFAULT_TAG,
+    NAMESPACE_COL,
+    NAME_COL,
+    TAG_COL,
+    PRIVATE_COL,
+    PROJ_COL,
+    N_SAMPLES_COL,
+    SUBMISSION_DATE_COL,
+    LAST_UPDATE_DATE_COL,
+    DIGEST_COL,
+    DB_TABLE_NAME,
+)
+from pepdbagent.utils import tuple_converter, registry_path_converter
+
+from pepdbagent.models import AnnotationModel, AnnotationReturnModel
+from pepdbagent.exceptions import RegistryPathError, ProjectNotFoundError
+
+_LOGGER = logging.getLogger("pepdbagent")
+
+
+class PEPDatabaseAnnotation:
+    """
+    Class that represents project Annotations in the Database.
+
+    While using this class, user can retrieve all necessary metadata about PEPs
+    """
+
+    def __init__(self, con: BaseConnection):
+        """
+        :param con: Connection to db represented by BaseConnection class object
+        """
+        self.con = con
+
+    def get(
+        self,
+        namespace: str = None,
+        name: str = None,
+        tag: str = None,
+        query: str = None,
+        admin: Union[List[str], str] = None,
+        limit: int = DEFAULT_LIMIT,
+        offset: int = DEFAULT_OFFSET,
+    ) -> AnnotationReturnModel:
+        """
+        Get project annotations.
+        There is 5 scenarios how to get project or projects annotations:
+            - provide name, namespace and tag. Return: project annotations of exact provided PK(namespace, name, tag)
+            - provide only namespace. Return: list of projects annotations in specified namespace
+            - Nothing is provided. Return: list of projects annotations in all database
+            - provide query. Return: list of projects annotations find in database that have query pattern.
+            - provide query and namespace. Return:  list of projects annotations find in specific namespace
+                that have query pattern.
+        :param namespace: Namespace
+        :param name: Project name
+        :param tag: tag
+        :param query: query (search string): Pattern of name, tag or description
+        :param admin: admin name (namespace), or list of namespaces, where user is admin
+        :param limit: return limit
+        :param offset: return offset
+        :return: pydantic model: AnnotationReturnModel
+        """
+        if all([namespace, name, tag]):
+            found_annotation = list(
```

Comment:
Hey @Khoroshevskyi, I've been testing this with `pephub` and this line is a bug. Calling `list()` on a model defined through `pydantic` does not create a list with one instance of the model in it, it actually creates a list of `tuples` from the key-value pairs in the model.

Everything inherits from the `BaseModel` class - check out the `__iter__` [method](https://github.com/pydantic/pydantic/blob/ef8f9fe51b9d0e38693f70482c29ba124f810d5a/pydantic/main.py#L488) of `BaseModel`. `list()` takes an iterable. I'd suggest just making this line:

```python
            found_annotation = [self._get_single_annotation(
                    namespace=namespace,
                    name=name,
                    tag=tag,
                    admin=admin,
                )
            )]
```

Reply:
do you mean changing `list()` to `[]`?

Reply:
Yeah. `list()` [takes an iterable](https://docs.python.org/3/library/functions.html#func-list) and generates a list from that.

Reply:
I see, I just got the same error. Fixing ...

Reply:
yes, you are right, you have to use `[]`